src/main/scala/analysis/SymbolicAccessAnalysis.scala

```diff
@@ -0,0 +1,151 @@
+ * Combination of reaching definitions and constant propagation
+ * elements in D are symbolic accesses of the form (variable, symbolic base, concrete offset)
+ * lattice L is a binary lattice with top and bottom
+ */
+trait SymbolicAccessFunctions(constProp: Map[CFGPosition, Map[Variable, FlatElement[BitVecLiteral]]]) extends ForwardIDEAnalysis[SymbolicAccess, TwoElement, TwoElementLattice] {
+
+  private val stackPointer = Register("R31", 64)
+  private val linkRegister = Register("R30", 64)
+  private val framePointer = Register("R29", 64)
+  private val mallocVariable = Register("R0", 64)
+
+  var mallocCount: Int = 0
+  private def nextMallocCount = {
+    mallocCount += 1
+    s"malloc_$mallocCount"
+  }
+
+  var unknownCount: Int = 0
+  private def nextunknownCount = {
+    unknownCount += 1
+    s"unknown_$unknownCount"
+  }
+
+  val valuelattice: TwoElementLattice = TwoElementLattice()
+  val edgelattice: EdgeFunctionLattice[TwoElement, TwoElementLattice] = EdgeFunctionLattice(valuelattice)
+  import edgelattice.{IdEdge, ConstEdge}
+
+
+  def edgesCallToEntry(call: DirectCall, entry: Procedure)(d: DL): Map[DL, EdgeFunction[TwoElement]] =
+    d match
+      case Left(value) =>
+        value.symbolicBase match
+          case StackLocation(regionIdentifier, parent, size) => Map()
+          case _ => Map(d -> IdEdge())
+      case Right(_) => Map(d -> IdEdge())
+
+  def edgesExitToAfterCall(exit: IndirectCall, aftercall: GoTo)(d: DL): Map[DL, EdgeFunction[TwoElement]] =
+    d match
+      case Left(value) =>
+        value.symbolicBase match
+          case StackLocation(regionIdentifier, parent, size) => Map()
+          case _ =>
+            if value.accessor.name == "R29" then
+              Map()
+            else Map(d -> IdEdge())
+      case Right(_) => Map(d -> IdEdge())
+
+  def edgesCallToAfterCall(call: DirectCall, aftercall: GoTo)(d: DL): Map[DL, EdgeFunction[TwoElement]] =
+    d match
+      case Left(value) =>
+        value.symbolicBase match
+          case StackLocation(regionIdentifier, parent, size) => Map(d -> IdEdge())
+          case _ => Map() // maps all variables before the call to bottom
+      case Right(_) => Map(d -> IdEdge())
+
+  def edgesOther(n: CFGPosition)(d: DL): Map[DL, EdgeFunction[TwoElement]] =
+    n match
+      case Assign(variable, rhs, maybeString: Option[String]) =>
+        val expr = unwrapPaddingAndSlicing(rhs)
+        expr match
+          case BinaryExpr(op, arg1: Variable, arg2) =>
+            evaluateExpression(arg2, constProp(n)) match
+              case Some(v) =>
+                if op.equals(BVADD) && arg1.equals(stackPointer) && v.value >= BITVECNEGATIVE then
+                  d match
+                    case Left(value) if value.accessor == variable => Map()
+                    case Left(value) => Map(d -> IdEdge())
+                    case Right(_) =>
+                      val size = twosComplementToDec(decToBinary(v.value))
```

Comment:
As far as I can tell this is just implementing conversion from bv64 to signed integer, it would be better to add this to bitVectorEval as `bv2int`, and maintain bitvector-width type-safety rather than assuming 64 bits. :)